### ISSUE TITLE
[All Source 1 Games][Critical] Spray Exploit

### ISSUE THREAD

**Issue by AhRoon** [ValveSoftware/Source-1-Games#3249]
Hello,

I found a bug regarding the spray exploit on All Source 1 Games. This time it is different with Crash Bot Issue. The others could crash the server by abusing Spray features using invalid spray.

**Comment by Pinsplash** [comment]
apparently this was just fixed, but only for tf2..?

**Comment by AhRoon** [comment]
Yeah the other games is not fixed. Hopefully valve patch the exploit on all games.

**Comment by CanadianJeff** [comment]
sourcemod offers a plugin that will scan detect and remove all invalid sprays from a server look on alliedmodders forums

**Comment by destoer** [comment]
This is still broken on css

**Comment by CanadianJeff** [comment]
valve should just open source all source engine 1

**Comment by CanadianJeff** [comment]
razzy — Today at 12:30 AM
a 128x128 render target with that is a depth buffer that doesnt have a depth buffer doesnt seem to make any sense, but it's the source code engine so :anything goes:™️

**Comment by alexiscoutinho** [comment]
> because the Tsuey issues pages removed all video links demoing the crash I thought I would put them back here
> 
> https://www.youtube.com/watch?v=rhzaKbmDg0Q https://streamable.com/pi8jus
> 
> these 2 videos are demos of L4D2 however several other games are also vuln using the same method

They can still be found in the comments' history though.

**Comment by CanadianJeff** [comment]
still cringe that Tsuey would remove the video links from his github I mean it is within his right todo so but if no one knows how this crash is done it will most likely never get fixed

**Comment by CanadianJeff** [comment]
can anyone confirm if garrys mod is vuln to this crashing spray?

**Comment by CanadianJeff** [comment]
pretty sure just like the official forums steam/valve does not care anymore

its been well over 1 year and still not patched

https://www.youtube.com/watch?v=yzbkUYvKOmg

**Comment by CanadianJeff** [comment]
I guess this is now patched in L4D2 still not confirmed if this is patched if any of the other source 1 based games I have personally notified to the Goldeneye Source devs about the sourcemod plugin and they have applied the fix

**Comment by nitro-1** [comment]
The issue is not fixed yet, today a new exploit is released and doing the same thing, crash players games...tested this exploit in left 4 dead 2.
[spray.zip](https://github.com/ValveSoftware/Source-1-Games/files/7774828/spray.zip)

**Comment by AshThe9thSurvivor** [comment]
I just saw a publication referring to L4D1, where it mentions this exploit, and as they always marked it as repeated, what's the point of doing this if they're not going to fix it, besides they always forget about L4D1, I've been reporting errors here several times and never I have been heard, the errors are still there, even requests that are not even reviewed, it seems to me.

**Comment by CanadianJeff** [comment]
has this crap ever been reported to hacker one???????

**Comment by CanadianJeff** [comment]
looks like Tsuey on the L4D2 thread has considered this issue closed but that does not mean it is not still an issue with other source based games so I will leave this issue open

**Comment by AndrengSG** [comment]
Can confirm this is still a prevalent issue in L4D1, even though it is fixed in L4D2.

**Comment by SharpieTheDergun** [comment]
If this is fixed in L4D2 then it's a new exploit with sprays. Because I just had a running Hunter in a Versus game spray a black texture in front of me and immediately crashed. Before that I crashed twice in two separate games.

**Comment by SharpieTheDergun** [comment]
Can confirm there's a new spray crash exploit going around. Pinsplash suggested looking into left4dead2/downloads folder as the .dat files as basically .vtf sprays. I've tested the spray in question and I can confirm it crashes your game. I'm going to upload it here as it's a critical issue.

https://www.dropbox.com/scl/fi/7hawlxh83hpq4mde2kcre/servercrash.vtf?rlkey=3dfkunjjb1jmf0nn6puzwv2op&st=9ig904r7&dl=0
